lib/ui/home/home_screen.dart

```diff
@@ -29,29 +60,52 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
   Widget build(BuildContext context) {
     ref.listen<HomeViewState>(homeViewModelProvider, (_, state) {
       state.maybeWhen(
-          getUserProfileSuccess: (profile) => {
-                ref.read(_profileAvatarProvider.notifier).state =
-                    profile.avatarUrl,
-              },
-          loading: () {
-            ref.read(_loadingStateProvider.notifier).state = true;
-          },
-          error: (message) => {
-                ref.read(_loadingStateProvider.notifier).state = false,
-                showToastMessage(message)
-              },
+          success: () => {},
+          loading: () {},
+          error: (message) => {showToastMessage(message)},
           orElse: () => {});
     });
+    bool isLoading = ref.watch(profileStream).value == null;
     return Scaffold(
-      body: SafeArea(
-        child: Consumer(
-          builder: (context, widgetRef, _) {
-            return HomeHeader(
-              date: DateTime.now().getFormattedString(),
-              avatar: widgetRef.watch(_profileAvatarProvider),
-            );
+        body: isLoading
+            ? const SurveyShimmerLoading()
+            : _buildHomeContent(mockSurveys));
+  }
```

Comment:
again, why don't you use `loading` state from `HomeViewState`? Something like:
```dart
return Scaffold(
    body: state.when(
        loading: () => const SurveyShimmerLoading(),
        success: () => _buildHomeContent(mockSurveys)),
        ....
        ),
    );
```

Reply:
@manh-t The ViewState is supposed to handle the initial loading instead, we will have to handle the survey loading as well. It will cause the View State changes frequently, that's why I split to handle in Stream instead.

Reply:
@hoangnguyen92dn I can see it in your integration PR:
![image](https://user-images.githubusercontent.com/60863885/230537305-c473da04-60ce-47f9-bc5a-b37f2fc30e36.png)
I think we can wait both `get profile` API and `get surveys` API done before changing the state. For example:
```
fetchData(){
    state = const HomeViewState.loading();
    final getProfileResult = await _getProfileUseCase.call();
    final getSurveysResult = await _getSurveyUseCase.call(input);
    if (...) {
        state = HomeViewState.error(error);
    } else {
        state = HomeViewState.success(error);
    }
}
```
Because currently I see you set the state to `loading`, `success`, or `error` but you don't use them, except `error` to show the toast message 🤔

Reply:
@manh-t There will be pullToRefresh so I expect to apply those states on this handler, how do you think?

Reply:
@hoangnguyen92dn when pulling to refresh, we only update the survey data so it doesn't relate to those states 🤔 
Please try to connect the `HomeViewState` to `HomeScreen` because the states in `HomeViewState` reflects the state of a screen:
- `loading` or `init`: show Shimmer loading
- `success`: show survey list
- `error`: show error screen or error message

Reply:
A good resource for `freezed`: https://codewithandrea.com/videos/flutter-state-management-setstate-freezed-state-notifier-provider/

Reply:
@manh-t 🤔 So for the View State we will handle like this:
- Init: Show shimmer loading 
- Loading: Pull to refresh
- Success: Return survey list
- Error: Show toast message

Reply:
@hoangnguyen92dn actually pull to refresh handle the loading itself and we haven't had this feature, so we don't need to count it here. I think we can remove `loading` usage for now, `init` is enough to show shimmer loading 🙏

Reply:
@manh-t 🙏 

> I will do refactor on the next PR, this PR is mainly for the UI Part.